### PR TITLE
feat: support a dedicated programmeName filter

### PR DESCRIPTION
**Pull request by JosephKelly**
This is the first of new filters.
A limitation means searching by optional partial terms aren't supported.

The new controller parameters should be cleaned up later. It's not clear how multiple programmes will be supported.

chore(smell): resolve codestyle issues

TIS21-2722: Filter by programme name

### CODE REVIEW

**Comment by JosephKelly** [comment]
Conscious that this might be blocking some work, I thought I'd give others the option to merge it already if they are confident about the implementation.  I didn't get around to doing some local testing today but intend to do so to boost my own confidence & familiarity with ES in recommendations.

File: application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java [hunk 1]
```diff
@@ -123,7 +114,7 @@ public void updateTrainee(final DoctorsForDbDto gmcDoctor) {
   }
 
   public void updateTraineeAdmin(final List<TraineeAdminDto> traineeAdmins) {
-    traineeAdmins.stream().forEach(traineeAdmin -> {
+    traineeAdmins.forEach(traineeAdmin -> {
```
Comment (jayanta2018) [inline application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java:117]: probably this will make the code faster

**Review by jayanta2018 — APPROVED** [review]
My local seems to be ok. LGTM

File: application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java [hunk 2]
```diff
@@ -91,16 +88,12 @@ public TraineeSummaryDto getAllTraineeDoctorDetails(final TraineeRequestDto requ
...
-        .traineeInfo(traineeDoctors)
-        .countTotal(getCountAll())
-        .countUnderNotice(getCountUnderNotice())
-        .totalPages(paginatedDoctors.getTotalPages())
-        .totalResults(paginatedDoctors.getTotalElements())
-        .build();
+    final var traineeDoctors = doctorsList.stream()
+        .map(recommendationViewMapper::toTraineeInfoDto).collect(toList());
+
+    return TraineeSummaryDto.builder().traineeInfo(traineeDoctors).countTotal(getCountAll())
```
Comment (CaiWillis) [inline application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java:94]: Might just be me but I find this less readable

Comment (JosephKelly) [reply application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java]: Changed this back

**Review by jayanta2018 — APPROVED** [review]

**Review by CaiWillis — APPROVED** [review]

**Pull request merged**